app/src/pages/inside/historyPage/historyView/historyTable/historyTable.jsx

```diff
@@ -196,38 +165,76 @@ export class HistoryTable extends Component {
   };
 
   renderHeader = () => {
-    const { history, intl } = this.props;
-    return history.map((historyItem) => (
-      <HistoryCell status={historyItem.launchStatus} key={historyItem.launchId} header>
-        {`${intl.formatMessage(messages.launchNumberTitle)}${historyItem.launchNumber}`}
-      </HistoryCell>
-    ));
+    const { intl } = this.props;
+    const headerItemsCount = this.getHeaderItemsCount();
+    const headerItems = [];
+    for (let index = headerItemsCount; index > 0; index -= 1) {
+      headerItems.push(
+        <HistoryCell key={index} header>
+          {`${intl.formatMessage(messages.executionNumberTitle)}${index}`}
+        </HistoryCell>,
+      );
+    }
+    return headerItems;
   };
+
+  renderHistoryItems = (item) => {
+    const itemLastIndex = this.rowItemsCount - 1;
+    const itemResources = [...item.resources].reverse();
+    const historyItems = [];
+
+    for (let index = itemLastIndex; index >= 0; index -= 1) {
```

Comment:
Maybe it would be better to slice `item.resources` array and transform it using `reverse` and `map`.

Reply:
Usually there are fewer elements in the `resources` array than the `itemLastIndex` value.
So no